src/entt/entity/mixin.hpp

```diff
@@ -22,16 +22,17 @@ namespace entt {
  *
  * This applies to all signals made available.
  *
- * @tparam Type The type of the underlying storage.
+ * @tparam Storage The type of the underlying storage.
+ * @tparam Registry The type of the underlying registry, defaulting to basic_registry
  */
-template<typename Type>
-class sigh_mixin final: public Type {
-    using underlying_type = Type;
+template<typename Storage, typename Registry>
+class sigh_mixin final: public Storage {
+    using underlying_type = Storage;
     using basic_registry_type = basic_registry<typename underlying_type::entity_type, typename underlying_type::base_type::allocator_type>;
```

Comment:
I think you want to update this member type and use it all over as it happened before already.
All other changes seem unnecessary once you updated this one. Am I wrong?

Reply:
It's needed in one place, in `bind`. The issue being that the `any value` passed there is of type `basic_registry`, not of the inherited one. And actually this part is the one that I'm not so certain about the approach, cause of the `reinterpret_cast` needed there